### ISSUE TITLE
ExceptionHandlerHelper always returns HTTP500 and code 113

### ISSUE THREAD

**Issue by nonovd** [MarcinOrlowski/laravel-api-response-builder#132]
ExceptionHandlerHelper always returns HTTP 500 status code and API code 113 with default config file (no mappings).

```php
{
    "success": false,
    "code": 113,
    "locale": "fr",
    "message": "Unauthenticated.",
    "data": null,
    "debug": {
        "trace": {
            "class": "Illuminate\\Auth\\AuthenticationException",
            "file": "/var/www/project/vendor/laravel/framework/src/Illuminate/Auth/Middleware/Authenticate.php",
            "line": 81
        }
    }
}
```

```php
{
    "success": false,
    "code": 113,
    "locale": "fr",
    "message": "No query results for model X",
    "data": null,
    "debug": {
        "trace": {
            "class": "Illuminate\\Database\\Eloquent\\ModelNotFoundException",
            "file": "/var/www/project/vendor/laravel/framework/src/Illuminate/Routing/ImplicitRouteBinding.php",
            "line": 38
        }
    }
}
```

**Expected behavior**
Is this correct behavior ? I thought it would handle basic framework exceptions and return appropriate HTTP codes out of the box.

**To reproduce**
In app/Exceptions/Handler.php
```php
    public function render($request, Exception $exception)
    {
        return ExceptionHandlerHelper::render($request, $exception);
    }
```
**Environemnt:**
 - PHP version 7.3
 - Laravel version 6.6.0
 - Laravel Response Builder 7.0.1

**Comment by MarcinOrlowski** [comment]
What class is `$exception` exactly?

**Comment by MarcinOrlowski** [comment]
Yes it works as designed. There's no config entry for this particular exception class, so default handler kicks in: https://github.com/MarcinOrlowski/laravel-api-response-builder/blob/master/src/ExceptionHandlerHelper.php#L58

**Comment by MarcinOrlowski** [comment]
Or let's edit this one

**Comment by nonovd** [comment]
I feel it should at least handle the following exceptions:

Exception | HTTP code
------------ | -------------
Illuminate\\Auth\\AuthenticationException | HttpResponse::HTTP_UNAUTHORIZED = 401
Illuminate\\Auth\\Access\\AuthorizationException | HttpResponse::HTTP_FORBIDDEN = 403
Illuminate\\Database\\Eloquent\\ModelNotFoundException | HttpResponse::HTTP_NOT_FOUND = 404

`use Symfony\Component\HttpFoundation\Response as HttpResponse;`

What do you think ?

**Comment by MarcinOrlowski** [comment]
Makes sense. The only concern I have about adding more to default settings are api codes. With defaults, i need to have own codes which means reserved range is still in use. From the other hand, having wider coverage in defaults makes packages easier to setup and use. Hmm

**Comment by nonovd** [comment]
Yes I understand. Maybe it is enough to put these exceptions as comments in the config file. Just to keep in mind which ones are not working out of the box.

```php
'exception_handler' => [
'map' => [
            HttpException::class => [
                HttpResponse::HTTP_SERVICE_UNAVAILABLE => [
                    'api_code' => ApiCodes::YOUR_API_CODE,
                ],
            ],
            Illuminate\Auth\AuthenticationException::class => [
                'api_code' => ApiCodes::YOUR_API_CODE,
                'http_code' => HttpResponse::HTTP_UNAUTHORIZED,
            ],
            Illuminate\Auth\Access\AuthorizationException::class => [
                'api_code' => ApiCodes::YOUR_API_CODE,
                'http_code' => HttpResponse::HTTP_FORBIDDEN,
            ],
            Illuminate\Database\Eloquent\ModelNotFoundException::class => [
                'api_code' => ApiCodes::YOUR_API_CODE,
                'http_code' => HttpResponse::HTTP_NOT_FOUND,
            ],
]
]
```

**Comment by MarcinOrlowski** [comment]
After a thought - this most likely needs a logic update as the more classes you add, inheritance may kick in and you'd love to have items prioritized, same as in `converter` configuration, otherwise it will not be too useful in case of bigger configs.

**Comment by MarcinOrlowski** [comment]
As I found some spare cycles, this feature is finally coming. This is still WiP but almost 95% baked already, so please take a look if you are still interested: https://github.com/MarcinOrlowski/laravel-api-response-builder/tree/delegated-exception-handlers

**Comment by MarcinOrlowski** [comment]
This feature is now merged into `dev` branch and will be released in v8.0.0 later this week.
